Mono.Addins/Mono.Addins/RuntimeAddin.cs

```diff
@@ -605,28 +605,42 @@ internal AddinDescription Load (Addin iad)
 			baseDirectory = description.BasePath;
 			module = description.MainModule;
 			module.RuntimeAddin = this;
-			
-			if (description.Localizer != null) {
-				string cls = description.Localizer.GetAttribute ("type");
-				
-				// First try getting one of the stock localizers. If none of found try getting the type.
-				object fob = null;
-				Type t = Type.GetType ("Mono.Addins.Localization." + cls + "Localizer, " + GetType().Assembly.FullName, false);
-				if (t != null)
-					fob = Activator.CreateInstance (t);
-				
-				if (fob == null)
-					fob = CreateInstance (cls, true);
-				
-				IAddinLocalizerFactory factory = fob as IAddinLocalizerFactory;
-				if (factory == null)
-					throw new InvalidOperationException ("Localizer factory type '" + cls + "' must implement IAddinLocalizerFactory");
-				localizer = new AddinLocalizer (factory.CreateLocalizer (this, description.Localizer));
+
+			// Register localizer types.
+			foreach (ExtensionNodeDescription localizerType in iad.Description.LocalizerTypes) {
+				var typeName = localizerType.GetAttribute ("type");
+
+				var registeredLocalizer = (IAddinLocalizer)CreateInstance (typeName, true);
```

Comment:
This is not right, the specified type is an implementation of IAddinLocalizerFactory.

Reply:
You're right, I'll take a look.

Reply:
Fixed.